### PR TITLE
Time handling in variant interface

### PR DESCRIPTION
**Pull request by merowin**
issue: #336 

I've made the changes suggested in https://github.com/govariantsteam/govariants/pull/335#issuecomment-2486948414
This adds a required field in the variant interface, but implementers can set it to "none" for starters.

On the server, I replaced the handler map by a function that uses the above.
The client does not displays the time config form when the above is set to "none".

### CODE REVIEW

File: packages/server/src/time-control/time-control.ts [hunk 1]
```diff
@@ -59,6 +62,31 @@ export function ValidateTimeControlBase(
   );
 }
 
+/**
+ * Returns a new time handler instance suited for a given variant.
+ */
+export function GetTimeHandler(
```
Comment (benjaminpjones) [inline packages/server/src/time-control/time-control.ts:68]: nit - we should probably figure out what we want our naming conventions to be, but usually functions are lowerCamel (whereas UpperCamel is reserved for classes and types)

**Review by benjaminpjones — APPROVED** [review]
Awesome! Glad to have one fewer map around

File: packages/server/src/time-control/time-control.ts (hunk 1 above)
Comment (merowin) [reply packages/server/src/time-control/time-control.ts]: Yeah I've been inconsistent, sorry about that. At work we have different technology stacks for frontend and backend, and the style guidelines also differ. So I'm used to e.g. write upper case in the backend.

Anyway, I renamed the function.

Comment (merowin) [reply packages/server/src/time-control/time-control.ts]: How do we want to write variables, by the way? I've seen Snake-Case, but I've also used Camel-Case.

Comment (benjaminpjones) [reply packages/server/src/time-control/time-control.ts]: I've been super inconsistent about this unfortunately. From a brief web search, it seems like lowerCamel might be more common for JS, generally.

- https://github.com/airbnb/javascript?tab=readme-ov-file#naming--camelCase
- https://google.github.io/styleguide/jsguide.html#naming-local-variable-names
- https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide/Code_style_guide/JavaScript#variables

**Pull request merged**